atom/src/lib.rs

```diff
@@ -171,6 +171,10 @@ where
     ) -> Option<Self::WriteHandle>;
 }
 
+pub trait BackAsSpace<'a>: Atom<'a, 'a> {
```

Comment:
I'm having trouble understanding what's going on.
Can you explain what this does and why it's used?

Reply:
Yeah, I haven't documented anything yet ^^"

The "back as space" operation is something that takes an atom's read handle and turns it back into a space (i.e. back to a bare atom buffer).

This is necessary because, when reading options from the plugin, the LV2 options spec requests a pointer to an atom (alongside the regular size and type).
This is a completely different operation than what is usually done from the Atom ports: we're not writing to or reading from a buffer from the host, instead the host is basically requesting a buffer _owned by the plugin_ with the value of the requested atom.

I've looked at implementations and usages of this, and the most usual case is options with scalars (mostly ints). In that case going from a read handle (`&i32` in the case of Atom::Int) to a byte slice is trivial, however many atom types cannot be easily created from native types without writing them to a buffer first, so those cannot support the back_as_space operation.

That's why I had to make it a separate trait, it's because not every atom type supports that operation. 🙂